### ISSUE TITLE
SC2181 should be a more visible error in `set -e` mode

### ISSUE THREAD

**Issue by ilyagr** [koalaman/shellcheck#2903]
#### For new checks and feature suggestions
- [?] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this

> It gives a style warning that undersells the seriousness of the problem. https://www.shellcheck.net/wiki/SC2181 mentions the problem with `set -e` in the middle of the text, but it is not emphasized and easy to miss.
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
        #!/bin/bash
	command -v appdmg >/dev/null 2>&1
	if [ $? -eq 0 ]; then
		RESOURCES="/tmp/resources.$$"
		cp -r "$BASEDIR/resources" "$RESOURCES"
		sed -i '' "s/\/tmp\/signed\/UTM.app/\/tmp\/signed.$$\/UTM.app/g" "$RESOURCES/appdmg.json"
		appdmg "$RESOURCES/appdmg.json" "$OUTPUT/UTM.dmg"
		rm -rf "$RESOURCES"
	else
		echo "appdmg not found, falling back to non-customized DMG creation"
		hdiutil create -fs HFS+ -srcfolder "$SIGNED/UTM.app" -volname "UTM" "$OUTPUT/UTM.dmg"
	fi
```

#### Here's what shellcheck currently says:

```
Line 3:
        if [ $? -eq 0 ]; then
             ^-- SC2181 (style): Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.
```

Note that this is a "style" warning, so the user might assume it's not serious. VS Code's Shellcheck extension shows style warnings as "informational", de-emphasizing them.

#### Here's what I wanted or expected to see:

Mark it as a "warning" or "error".

```
Line 3:
        if [ $? -eq 0 ]; then
             ^-- SC2181 (warning): This if will never be executed if the previous statement results in a non-zero error code because of `set -e`. Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.
```
